accepted/benchmark-infrastructure.md

```diff
@@ -0,0 +1,102 @@
+# Summary
+[summary]: #summary
+
+This RFC proposes setting up a benchmarking and performance-tracking
+infrastructure to track Cranelift and wasmtime performance over time. We would
+like to (i) establish some server infrastructure, (ii) build automation
+integrated with our GitHub repos, and (iii) provide a web-based frontend to
+show historical and current performance data.
+
+# Motivation
+[motivation]: #motivation
+
+As we develop the Cranelift compiler and its machine backends, and as it is
+used in more production settings, its performance becomes increasingly
+important. In many settings, a few percentage points change in compilation time
+or generated-code performance is quite a big deal. In addition, experience in
+many compiler projects over the years has shown that small performance deltas
+have a way of accumulating over time; if there is no systematic tracking, this
+gradual drift can add up to sizeable performance regressions.
+
+It's thus important both to track historical trends, in order to avoid
+regressions and (more positively) to measure our progress; and to measure the
+effect of individual changes, when significant enough, to determine objectively
+whether optimizations or design choices are good ideas.
+
+Independently of *what* benchmarks we measure, we need an *infrastructure* to
+measure, record, and visualize statistics. This RFC proposes building such an
+infrastructure.
+
+# Proposal sketch
+[proposal]: #proposal
+
+## Step 1: Establish physical infrastructure
+
+First, we need to establish a set of machines to run benchmarks and host
+results. The GitHub-hosted CI infrastructure we currently use for testing will
+likely not be adequate, because (i) runtime of benchmarks will be longer than
+we would like for CI actions, (ii) we don't want to tie benchmark runs to CI
+runs, necessarily, and (iii) CI-runner environments are unpredictable and
+unreliable environments for accurate, low-noise benchmark measurements.
+
+We already have access to one AArch64 server-grade host for interactive
+development of the Cranelift ARM backend, generously [provided by
+ARM](https://github.com/WorksOnArm/cluster/issues/204). We may be able to use
+this host for running benchmarks as well, if all stakeholders agree.
+
+We will need an x86-64 host as well; the best option is likely to rent a
+low-cost dedicated host. Details of hosting choice and funding are TBD.
```

Comment:
FWIW this is what rust-lang/rust does and I think it's worked our relatively well for them in terms of reliability and cost.